lib/devise/models/confirmable.rb

```diff
@@ -44,7 +44,7 @@ module Confirmable
       included do
         before_create :generate_confirmation_token, if: :confirmation_required?
         after_create :skip_reconfirmation_in_callback!, if: :send_confirmation_notification?
-        if respond_to?(:after_commit) # ActiveRecord
+        if defined?(ActiveRecord::Base) # ActiveRecord
```

Comment:
Neither respond_to?(:after_commit), nor defined?(ActiveRecord::Base) are not working. What I found is:
```
if self.is_a?(ActiveRecord::Base)
```

Reply:
Makes sense. Using the `defined?` could also bring some problems for projects that use both `mongoid` and `active_record` at the same time.